setup/src/main/java/io/jenkins/jenkinsfile/runner/CliLauncher.java

```diff
@@ -0,0 +1,70 @@
+package io.jenkins.jenkinsfile.runner;
+
+import hudson.cli.CLICommand;
+import hudson.security.ACL;
+import io.jenkins.jenkinsfile.runner.bootstrap.Bootstrap;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * Sets up a Jenkins environment that provides an interactive CLI.
+ */
+public class CliLauncher extends JenkinsLauncher {
+    public CliLauncher(Bootstrap bootstrap) {
+        super(bootstrap);
+    }
+
+    @Override
+    protected int doLaunch() throws Exception {
+        // so that the CLI has all the access to the system
+        ACL.impersonate(ACL.SYSTEM);
+        BufferedReader commandIn = new BufferedReader(new InputStreamReader(System.in));
+        String line;
+        System.out.printf("Connected to Jenkins!%nType 'help' for a list of available commands, or 'exit' to quit.%n");
+        System.out.print(" > ");
+        while ((line = commandIn.readLine()) != null) {
+            if(line.equalsIgnoreCase("exit")) {
+                break;
+            } else if(line.isEmpty()) {
+                continue;
+            }
+            tryRunCommand(line);
+            System.out.print(" > ");
+        }
+        System.out.println("bye");
+        return 0;
+    }
+
+    private void tryRunCommand(String commandLine) {
+        try {
+            String[] parts = commandLine.split(" ", 2);
+            CLICommand command = CLICommand.clone(parts[0]);
+            if(command == null) {
+                System.err.println("No such command, try 'help'.");
+                return;
+            }
+            command.main(findArgsFromLineParts(parts), Locale.getDefault(), System.in, System.out, System.err);
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.err.println("An unexpected error occurred executing this command.");
+        }
+    }
+
+    private List<String> findArgsFromLineParts(String[] commandLineParts) {
```

Comment:
Could it be generalized in `JenkinsLauncher`

Reply:
I can move it there, but I don't see how this internal utility method belongs there semantically - the only other existing implementation doesn't do command-line parsing and this use-case seems very specific to this one mode.